tests/common/plugins/psu_controller/snmp_psu_controllers.py

```diff
@@ -9,40 +9,90 @@
 from controller_base import run_local_cmd
 
 
-def get_psu_controller_type(psu_controller_host):
+try:
+    from pysnmp.proto import rfc1902
+    from pysnmp.entity.rfc3413.oneliner import cmdgen
+    from pyasn1.type import univ
+    has_pysnmp = True
+except:
+    has_pysnmp = False
+
+class snmpPsuController(PsuControllerBase):
     """
-    @summary: Use SNMP to get the type of PSU controller host
-    @param psu_controller_host: IP address of PSU controller host
-    @return: Returns type string of the specified PSU controller host
-    """
-    result = None
-    cmd = "snmpget -v 1 -c public -Ofenqv %s .1.3.6.1.2.1.1.1.0" % psu_controller_host
-    try:
-        stdout = run_local_cmd(cmd)
-
-        lines = stdout.splitlines()
-        if len(lines) > 0:
-            result = lines[0].strip()
-            result = result.replace('"', '')
-    except Exception as e:
-        logging.debug("Failed to get psu controller type, exception: " + repr(e))
-
-    return result
-
+    PSU Controller class for SNMP conrolled PSUs - 'Sentry Switched CDU' and 'APC Web/SNMP Management Card'
 
-class SentrySwitchedCDU(PsuControllerBase):
+    This class implements the interface defined in PsuControllerBase class for SNMP conrtolled PDU type 
+    'Sentry Switched CDU' and 'APC Web/SNMP Management Card'
     """
-    PSU Controller class for 'Sentry Switched CDU'
 
-    This class implements the interface defined in PsuControllerBase class for PDU type 'Sentry Switched CDU'
-    """
-    PORT_NAME_BASE_OID = ".1.3.6.1.4.1.1718.3.2.3.1.3.1"
-    PORT_STATUS_BASE_OID = ".1.3.6.1.4.1.1718.3.2.3.1.5.1"
-    PORT_CONTROL_BASE_OID = ".1.3.6.1.4.1.1718.3.2.3.1.11.1"
-    STATUS_ON = "1"
-    STATUS_OFF = "0"
-    CONTROL_ON = "1"
-    CONTROL_OFF = "2"
+    def get_psu_controller_type(self):
+        """
+        @summary: Use SNMP to get the type of PSU controller host
+        @param psu_controller_host: IP address of PSU controller host
+        @return: Returns type string of the specified PSU controller host
+        """
+        pSYSDESCR = ".1.3.6.1.2.1.1.1.0"
+        SYSDESCR = "1.3.6.1.2.1.1.1.0"
+        psu = None
+        if not has_pysnmp:
+            logging.info("Missing required pysnmp module (check docs)")
+        else:
+            cmdGen = cmdgen.CommandGenerator()
+            snmp_auth = cmdgen.CommunityData('public')
```

Comment:
looks like hard coded data? I suggest to the creds into group_vars/all/psu.yml and then use creds fixture to load it.

Reply:
will fix this with separate PR.